### PR TITLE
[14.0][ADD] product_multi_barcode_constraint_per_company: add module

### PR DESCRIPTION
**Pull request by dessanhemrayev**
This is a bridge module between "product_multi_barcode" and "product_barcode_constraint_per_company"

### CODE REVIEW

**Comment by dessanhemrayev** [comment]
Following this @legalsylvain 's [comment](https://github.com/OCA/stock-logistics-barcode/pull/537#issuecomment-1778062876).

File: product_multi_barcode_constraint_per_company/models/product_barcode.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from odoo import api, models
+
+
+class ProductBarcode(models.Model):
+    _inherit = "product.barcode"
+
+    @api.constrains("name")
+    def _check_duplicates(self):
+        for record in self:
+            barcodes = self.search(
```
Comment (geomer198) [inline product_multi_barcode_constraint_per_company/models/product_barcode.py:13]: Could you add a limit parameter?
barcodes = self.search([...], limit=1)

File: product_multi_barcode_constraint_per_company/tests/test_product_barcode_constraint.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+from odoo.addons.product_barcode_constraint_per_company.tests.test_module import (
+    TestModule,
+)
+
+from .. import void
+
+
+@tagged("post_install", "-at_install")
+class TestProductBarcodeConstraintInherit(TestModule):
```
Comment (geomer198) [inline product_multi_barcode_constraint_per_company/tests/test_product_barcode_constraint.py:18]: For each tests need docstring and assertion message.

**Review by geomer198 — COMMENTED** [review]
Please fix comments

File: product_multi_barcode_constraint_per_company/models/product_barcode.py (hunk 1 above)
Comment (dessanhemrayev) [reply product_multi_barcode_constraint_per_company/models/product_barcode.py:13]: Thanks, I'll fix it now

File: product_multi_barcode_constraint_per_company/tests/test_product_barcode_constraint.py (hunk 2 above)
Comment (dessanhemrayev) [reply product_multi_barcode_constraint_per_company/tests/test_product_barcode_constraint.py:18]: Thanks, I'll fix it now

File: product_multi_barcode_constraint_per_company/readme/newsfragmensts/2944.feature.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+This is a bridge module between "product_barcode_constraint_per_company" and "product_multi_barcode"
```
Comment (geomer198) [inline product_multi_barcode_constraint_per_company/readme/newsfragmensts/2944.feature.rst]: Should be remove this file. News fragments using for updating module not for creating.

**Review by geomer198 — CHANGES_REQUESTED** [review]

**Review by geomer198 — APPROVED** [review]
LGTM!

**Pull request closed without merging**